### PR TITLE
Miscellaneous addendums to the {analyze,simplify}Expr() work.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: sql/analyze_test.go [hunk 1]
```diff
@@ -358,7 +360,7 @@ func TestSimplifyAndExprCheck(t *testing.T) {
 			t.Errorf("%s: expected %s, but found %s", d.expr, d.expected, s)
 		}
 
-		if d.check && !checkEquivExpr(t, expr1, expr2, qvals) {
+		if d.checkEquiv && !checkEquivExpr(t, expr1, expr2, qvals) {
```
Comment (tamird) [inline sql/analyze_test.go]: I think `checkEquivExpr` should return an error, which will make this handling way clearer

File: sql/analyze_test.go [hunk 2]
```diff
@@ -147,6 +147,8 @@ func TestSimplifyExpr(t *testing.T) {
 
 		{`a IN (1, 1)`, `a IN (1)`},
 		{`a IN (2, 3, 1)`, `a IN (1, 2, 3)`},
+		{`a IN (1, true)`, `true`},
```
Comment (tamird) [inline sql/analyze_test.go]: what's this testing? this expression shouldn't be valid

Comment (tamird) [inline sql/analyze_test.go]: just read the comment on `typeCheckTuple`; I guess producing an error from this input will come later.

File: sql/analyze.go [hunk 3]
```diff
@@ -874,6 +874,10 @@ func simplifyOneOrExpr(left, right parser.Expr) (parser.Expr, parser.Expr) {
 }
 
 func simplifyOneOrInExpr(left, right *parser.ComparisonExpr) (parser.Expr, parser.Expr) {
+	if left.Operator != parser.In && right.Operator != parser.In {
+		panic(fmt.Sprintf("IN expression required: %s vs %s", left, right))
```
Comment (tamird) [inline sql/analyze.go]: the panic below is now unreachable

**Comment by tamird** [comment]
LG

File: sql/analyze_test.go (hunk 2 above)
Comment (petermattis) [inline sql/analyze_test.go]: Yeah, the expression isn't valid. Prior to `typeCheckTuple` sorting of the values would panic due to int and bool not being comparable. I think we need a `TypeCheckExpr` pass that is run before simplification.

File: sql/analyze.go (hunk 3 above)
Comment (petermattis) [inline sql/analyze.go]: Good point. Removed.

File: sql/analyze_test.go (hunk 1 above)
Comment (petermattis) [inline sql/analyze_test.go]: Heh, well I thought it was clear. Changed `checkEquivExpr` to return an error.

**Pull request merged**